org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/SwitchPatternTest.java

```diff
@@ -2651,11 +2651,6 @@ public void testBug573921_7() {
 				"	case List<String> s: \n" +
 				"	     ^^^^^^^^^^^^^^\n" +
 				"Type Object cannot be safely cast to List<String>\n" +
-				"----------\n" +
-				"3. ERROR in X.java (at line 9)\n" +
-				"	case List<String> s: \n" +
-				"	     ^^^^^^^^^^^^^^\n" +
-				"This case label is dominated by one of the preceding case labels\n" +
 				"----------\n");
```

Comment:
I gather this diagnostic going away is because of the early exit in CaseStatement ? If so, OK

Reply:
Right, when `isApplicable()` returns false, we've seen enough, `resolveCasePattern()` returns `NotAConstant`, hence no `ResolvedCase` is created and `SwitchStatement.resolve()` has nothing to check dominance for.